File: src/main/java/com/ordnance/karakal/websocket/messages/DrawMessage.java
package com.ordnance.karakal.websocket.messages;

import com.ordnance.karakal.websocket.DRAW;

import java.util.List;
import java.util.UUID;

public class DrawMessage implements ClientMessage{
    public UUID gameId;
    public DRAW drawType;
    public UUID playerId;
    public Long cardId;
}

File: frontend/src/utilities/websocket/messages/DrawMessage.ts
export type DrawMessage = {
    type: "DRAW",
    gameId: number,
    drawType: 'DECK' | 'DISCARD',
    playerId: string,
    cardId? : number
}

export const createDrawMessage = (gameId: string, drawType : 'DECK' | 'DISCARD', playerId: string, cardId? : number) => ({
    type: "DRAW",
    gameId,
    drawType,
    playerId,
    cardId
})